Resources/public/js/views/ez-userprofileview.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-userprofileview', function (Y) {
+    'use strict';
+    /**
+     * Provides the User Profile View class
+     *
+     * @module ez-userprofileview
+     */
+    Y.namespace('eZ');
+
+    var CLASS_USER_PROFILE_HIDDEN = 'is-user-profile-hidden',
+        SELECTOR_USER_NAME = '.ez-user-profile-name',
+        SELECTOR_USER_ROLE = '.ez-user-profile-role',
+        SELECTOR_USER_IMAGE = '.ez-user-profile-image';
+
+    /**
+     * The User Profile View
+     *
+     * @namespace eZ
+     * @class UserProfileView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.UserProfileView = Y.Base.create('userProfileView', Y.eZ.TemplateBasedView, [], {
+        events: {
+            '.ez-user-profile': {
+                'tap': '_toggleUserMenu'
+            }
+        },
+
+        initializer: function () {
+           this.after('*:userInfoChange', this._setUserInfo, this);
+        },
+
+        /**
+        * Renders the user profile view
+        *
+        * @method render
+        * @return {eZ.UserProfileView} the view itself
+        */
+        render: function () {
+            this.get('container')
+                .setHTML(this.template())
+                .addClass(CLASS_USER_PROFILE_HIDDEN);
+
+            this.fire('getUserInfo');
+            this._renderUserMenu();
+
+            return this;
+        },
+
+        /**
+        * Renders the user menu view
+        *
+        * @method _renderUserMenu
+        */
+        _renderUserMenu: function () {
+            var view = this.get('userMenuView');
+
+            view.addTarget(this);
+            this.get('container').append(view.render().get('container'));
+            view.set('active', true);
+        },
+
+        /**
+         * Sets name, role and image for the user
+         *
+         * @method _setUserInfo
+         * @protected
+         */
+        _setUserInfo: function () {
+            var user = this.get('userInfo'),
+                container = this.get('container'),
+                fields = user.userData.Version.Fields.field,
+                imageUrl,
+                imageAltText;
+
+                fields.forEach(function (field) {
+                    if (field.fieldDefinitionIdentifier === 'image' && field.fieldValue) {
```

Comment:
there are several issues in this method:

1. it's a bad practice to base some logic on the `fieldDefinitionIdentifier` because a user is a Content item with a Content Type that might be different from what you expect. To do that in a reliable way, you should get the field definition identifier of an `ezimage` field def. from the Content Type with [`getFieldDefinitionIdentifiers`](http://ezsystems.github.io/platformui-javascript-api/classes/eZ.ContentType.html#method_getFieldDefinitionIdentifiers) (of course it's also possible for the user content type to not have any `ezimage` field.)
1. to properly get the field and its value, you should use the Content model
1. to display the user's avatar, you use the original image variation which can be quite big while it's displayed in a very small way. Also the img is styled to be a square (why in em ?), while the image might not have such ratio. So instead, you use a dedicated image variation to have a square generated from the image

so for these reasons, the code should be organized in a different way to have less logic here and more spread over the models and the view services. So basically, here is the changes you should do:

1. change the user model to add a `loadAvatar` method. Internally, this method will first load the corresponding Content Type, check if it has an `ezimage` field then load the corresponding Content item and retrieve the `ezimage` field:
  ```js
  user.loadAvatar({api: capiInstance}, function (err, field) {
      // field can be undefined if the user Content Type does not have any ezimage field def.
      // otherwise it's the field value of the first ezimage field.
  });
  ```
2. in the navigation hub view service, from the field, you should [load a specific image variation](http://ezsystems.github.io/javascript-rest-client/classes/ContentService.html#method_loadImageVariation) defined in image.yml which is then pass to the view in `getViewParameters`